packages/docusaurus-theme-classic/src/theme/BlogArchivePage/index.tsx

```diff
@@ -19,6 +21,17 @@ type YearProp = {
 };
 
 function Year({year, posts}: YearProp) {
+  const {
+    i18n: {currentLocale, localeConfigs},
+  } = useDocusaurusContext();
+
+  const formattedDate = (date: string) =>
+    formatBlogPostDate(
+      currentLocale,
+      date,
+      localeConfigs[currentLocale]!.calendar,
+    );
```

Comment:
This is too much "duplicated" complexity to add to many components. We need a better abstraction in theme-common than the `formatBlogPostDate` helper